packages/tupaia-web/src/api/queries/useDashboards.ts

```diff
@@ -13,11 +13,12 @@ export const useDashboards = (
   entityCode?: EntityCode,
   dashboardName?: DashboardName,
 ) => {
-  const { data = [], isLoading } = useQuery(
+  const enabled = !!entityCode && !!projectCode;
+  const { data = [], isLoading, isError } = useQuery(
     ['dashboards', projectCode, entityCode],
     (): Promise<TupaiaWebDashboardsRequest.ResBody> =>
       get(`dashboards/${projectCode}/${entityCode}`),
-    { enabled: !!entityCode && !!projectCode },
+    { enabled, keepPreviousData: enabled },
```

Comment:
This is so that if the query is now disabled, the old data does not remain visible

Reply:
thanks - I was perhaps too aggressive when I added keepPreviousData as true globally since it seems to be creating bugs. Let's keep an eye on it and if we have to keep adding exceptions we can turn off that default.